### DISCUSSION TITLE
Why is there 'LIVEMEDIA_API' at the head of all the classes

### DISCUSSION THREAD

**Discussion by Skwangles** [melchi45/live555#7] [Q&A] [answered]
The original live555 source code does not have the LIVEMEDIA_API value at the head of all the class declarations, just wondering if there is a purpose to adding this - as it seems to cause compile errors (mitigated by '#define LIVEMEDIA_API').

Thanks

**Comment by melchi45** [comment, marked as answer]
The LIVEMEDIA_API is dll export definition for dynamic link library  for windows environment. It should be need to binary linking for live555.dll. This code is not use the genWindowsMakefile or config.xxx file to generate for Makefile. You have to use the cmake for configure and building. Reccommand to read the Readme.md file.
